ISL.Providers.Storages.AzureBlobStorage.Tests.Unit/Services/Foundations/Files/StorageTests.Validations.CreateAndAssignAccessPolicies.cs

```diff
@@ -0,0 +1,96 @@
+﻿// ---------------------------------------------------------
+// Copyright (c) North East London ICB. All rights reserved.
+// ---------------------------------------------------------
+
+using FluentAssertions;
+using ISL.Providers.Storages.AzureBlobStorage.Models.Foundations.Files.Exceptions;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ISL.Providers.Storages.AzureBlobStorage.Tests.Unit.Services.Foundations.Files
+{
+    public partial class StorageTests
+    {
+        [Theory]
+        [MemberData(nameof(InvalidPolicyArguments))]
+        public async Task
+            ShouldThrowValidationExceptionOnCreateAccessPolicyIfArgumentsInvalidAsync(
+            string invalidText, List<string> invalidList)
+        {
+            // given
+            string invalidContainer = invalidText;
+            List<string> invalidPolicyNames = invalidList;
```

Comment:
It will probably be a bit easier to see what this test does if you change it to this.  And then duplicat the test to specifically check the other 2 conditions on invalidPolicyNames when `List<string> invalidPolicyNames = null;` or `List<string> invalidPolicyNames = new List<string>();`

```suggestion
        [Theory]
        [InlineData(null)]
        [InlineData("")]
        [InlineData(" ")]

        public async Task ShouldThrowValidationExceptionOnCreateAccessPolicyIfArgumentsInvalidAsync(string invalidText)
        {
            // given
            string invalidContainer = invalidText;
            List<string> invalidPolicyNames = new List<string> { invalidList};
```

Reply:
This has been updated